### PR TITLE
feat: release plan template strategy types, constraints, segments

### PR DESCRIPTION
**Pull request by daveleek**
Adds different types of strategies and their properties to the strategy-add/edit dialog, along with constraints and segments

Skjermbilde 2024-11-26 kl  15 25 55
Skjermbilde 2024-11-26 kl  15 25 36

### CODE REVIEW

**Comment by daveleek** [comment]
There's some validation to add, but that will be a separate task

File: frontend/src/component/releases/ReleasePlanTemplate/MilestoneStrategyType.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    if (!strategyDefinition) {
+        return null;
+    }
+
+    switch (strategy.name) {
+        case 'default':
+            return <DefaultStrategy strategyDefinition={strategyDefinition} />;
+        case 'flexibleRollout':
+            return (
+                <MilestoneStrategyTypeFlexible
```
Comment (daveleek) [inline frontend/src/component/releases/ReleasePlanTemplate/MilestoneStrategyType.tsx:32]: This one in feature strategies uses projectid and environment info, so had to take the relevant parts out and make a separate one specifically for this.

File: frontend/src/component/releases/ReleasePlanTemplate/MilestoneStrategyType.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    parameters={parameters}
+                    updateParameter={updateParameter}
+                    errors={errors}
+                    editable={true}
+                />
+            );
+        case 'userWithId':
+            return (
+                <UserWithIdStrategy
```
Comment (daveleek) [inline frontend/src/component/releases/ReleasePlanTemplate/MilestoneStrategyType.tsx:41]: This along with the other UIs (Default and General) are all imported from the feature strategies. Should we make our own here or is this good enough for now?

File: frontend/src/component/releases/ReleasePlanTemplate/ReleasePlanTemplateAddStrategyForm.tsx [hunk 3]
```diff
@@ -127,9 +191,35 @@ export const ReleasePlanTemplateAddStrategyForm = ({
                     )}
                 </StyledTitle>
             </StyledHeaderBox>
+            {!BuiltInStrategies.includes(strategy.name || 'default') && (
+                <StyledAlertBox>
+                    <Alert severity='warning'>
+                        Custom strategies are deprecated. We recommend not
```
Comment (daveleek) [inline frontend/src/component/releases/ReleasePlanTemplate/ReleasePlanTemplateAddStrategyForm.tsx:197]: Given that custom strategies are deprecated, should we support them here at all?

File: frontend/src/component/releases/ReleasePlanTemplate/MilestoneStrategyType.tsx (hunk 2 above)
Comment (nunogois) [reply frontend/src/component/releases/ReleasePlanTemplate/MilestoneStrategyType.tsx:41]: This is probably good enough for now, let's come back to this later.

File: frontend/src/component/releases/ReleasePlanTemplate/ReleasePlanTemplateAddStrategyForm.tsx (hunk 3 above)
Comment (nunogois) [reply frontend/src/component/releases/ReleasePlanTemplate/ReleasePlanTemplateAddStrategyForm.tsx:197]: IMO we can do the same as we do in feature strategies for now and return to this later. But it's good to bring it up.

**Review by nunogois — APPROVED** [review]
👍 Following the philosophy of moving forward for now and coming back to clean this up later.

**Pull request merged**